roles/network_plugin/cilium/templates/cilium/cr.yml.j2

```diff
@@ -28,6 +28,7 @@ rules:
   - pods
   - endpoints
   - nodes
+  - secrets
```

Comment:
AFAICT the secrets permission in the helm charts is only `get` for secrets, should this be as tight here ?

Reply:
I see the following errors in Cilium when i dont give watch/list

When I only have get permissions for secrets, I get the following Warn:
```
time="2024-09-05T18:09:04Z" level=warning msg="github.com/cilium/cilium/pkg/k8s/resource/resource.go:808: failed to list *v1.Secret: secrets is forbidden: User \"system:serviceaccount:kube-system:cilium\" cannot list resource \"secrets\" in API group \"\" in the namespace \"kube-system\"" subsys=klog
time="2024-09-05T18:09:04Z" level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/resource/resource.go:808: Failed to watch *v1.Secret: failed to list *v1.Secret: secrets is forbidden: User \"system:serviceaccount:kube-system:cilium\" cannot list resource \"secrets\" in API group \"\" in the namespace \"kube-system\"" subsys=k8s
```

When I only have get,list permissions for secrets, I get the following Warn:
```time="2024-09-05T18:11:59Z" level=error msg=k8sError error="github.com/cilium/cilium/pkg/k8s/resource/resource.go:808: Failed to watch *v1.Secret: unknown (get secrets)" subsys=k8s
```

Reply:
Hi @foobaar, Did you see the errors in cilium-agent pod?  Look at https://github.com/cilium/cilium/blob/c9723a8df3cfa336da1f8457a864105d8349acfe/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml#L66, I don't see that these extra list/watch permissions are needed cilium upstream, maybe we need to investigate further.

Reply:
I did/do see these errors in the cilium-agent pod.
The log messages I posted are from the cilium agent pod without list/watch permissions.
The version of cilium I am using is the current default for kubespray which is: 1.15.4

Reply:
I can remove the list/watch secrets from the MR if this is delaying/blocking this MR. 
Get secrets and ciliumloadbalancerippools are required by the agent though.

Reply:
Yep. I am defintely using the kubespray defaults.
```
root@k8sdata30:/home/cilium# cilium version
Client: 1.15.4 9b3f9a8c 2024-04-11T17:25:42-04:00 go version go1.21.9 linux/amd64
Daemon: 1.15.4 9b3f9a8c 2024-04-11T17:25:42-04:00 go version go1.21.9 linux/amd64
```

Reply:
Like I mentioned, I can go ahead and remove watch/list. But i do want to note the warns in the logs that I saw.

Reply:
@foobaar Is this a problem with version 1.15.8? If it still exists, you should report it to the Cilium issue. Otherwise, we'll update Cilium to version 1.15.8.

Reply:
Updated the PR to only get secrets and ciliumloadbalancerippools.
These two are missing in cluster role for cilium in kubespray

Reply:
As for the list/watch, i will report that to cilium in a separate issue.

Reply:
> Updated the PR to only get secrets and ciliumloadbalancerippools.
These two are missing in cluster role for cilium in kubespray

Yes, It's better to report that to cilium in a separate issue, we should keep consites with the upstream.

Reply:
Sounds good! I will so that. Thanks @cyclinder

Reply:
I have updated the MR

Reply:
Hi @foobaar, Do you open this issue to cilium repo?

Reply:
I could not submit an issue for the current version of cilium 1.15.4 https://github.com/kubernetes-sigs/kubespray/blob/15bb5b0789af1d8e1cdee2ff1aa80d0d623d6653/README.md?plain=1#L171

To submit a bug report, i am required to be on atleast 1.15.7

Reply:
I will upgrade and submit an issue to cilium this week